sound/soc/sof/ipc4-topology.c

```diff
@@ -846,6 +857,57 @@ static void sof_ipc4_widget_free_comp_src(struct snd_sof_widget *swidget)
 	swidget->private = NULL;
 }
 
+static int sof_ipc4_widget_setup_comp_asrc(struct snd_sof_widget *swidget)
+{
+	struct snd_soc_component *scomp = swidget->scomp;
+	struct sof_ipc4_asrc *asrc;
+	int ret;
+
+	dev_dbg(scomp->dev, "Updating IPC structure for %s\n", swidget->widget->name);
+
+	asrc = kzalloc(sizeof(*asrc), GFP_KERNEL);
+	if (!asrc)
+		return -ENOMEM;
+
+	swidget->private = asrc;
+
+	ret = sof_ipc4_get_audio_fmt(scomp, swidget, &asrc->available_fmt, &asrc->base_config);
+	if (ret)
+		goto err;
```

Comment:
is this label correct? That would result in freeing something that wasn't properly allocated, is this intentional?

Or do we need to labels to jump directly to line 895 and skip the sof_ipc4_free_audio_fmt() call